paasta_tools/deployd/common.py

```diff
@@ -1,31 +1,61 @@
 import logging
 import time
 from collections import namedtuple
+from queue import PriorityQueue
 from queue import Queue
 from threading import Thread
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
 from paasta_tools.marathon_tools import get_all_marathon_apps
 from paasta_tools.marathon_tools import get_marathon_client
 from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.marathon_tools import load_marathon_service_config_no_cache
 from paasta_tools.utils import InvalidJobNameError
 from paasta_tools.utils import NoDeploymentsAvailable
 from paasta_tools.utils import NoDockerImageError
 
 BounceTimers = namedtuple('BounceTimers', ['processed_by_worker', 'setup_marathon', 'bounce_length'])
-ServiceInstance = namedtuple(
+ServiceInstanceTuple = namedtuple(
     'ServiceInstance', [
         'service',
         'instance',
         'bounce_by',
         'watcher',
         'bounce_timers',
         'failures',
+        'priority',
     ],
 )
 
 
+def ServiceInstance(service, instance, watcher, cluster, bounce_by, failures=0, bounce_timers=None, priority=None):
```

Comment:
to me it looks like you actually just want to define the initialization:

```python

class NT(collections.namedtuple('NT', ('foo', 'bar'))):
    # restore immutability that subclassing breaks
    __slots__ = ()

    def __new__(cls, foo, bar=5):
        # or with more logic
        return cls(foo, bar)
```

Note that (as with most immutable objects) you must override `__new__` instead of `__init__` (as `__init__` is too late)